### PR TITLE
Re-enable LAN867x option in ethernet_init

### PR DESCRIPTION
**Pull request by bogdankolendovskyy**
Re-enables an option to use LAN867x in ethernet_init

### CODE REVIEW

File: common_examples/simple-ethernet/main/idf_component.yml [hunk 1]
```diff
@@ -4,4 +4,8 @@ dependencies:
   ethernet_init:
     version: '*'
     # For local development use the local copy of the component
-    override_path: '../../../ethernet_init/'
\ No newline at end of file
+    override_path: '../../../ethernet_init'
+  espressif/lan867x:
```
Comment (kostaond) [inline common_examples/simple-ethernet/main/idf_component.yml:8]: Why need to add LAN867x to examples? Shouldn't it be pulled by `Ethernet Init` component?

File: ethernet_init/idf_component.yml [hunk 2]
```diff
@@ -1,4 +1,7 @@
 dependencies:
```
Comment (kostaond) [inline ethernet_init/idf_component.yml:1]: Please update the minor version of the component.

File: common_examples/simple-ethernet/main/idf_component.yml (hunk 1 above)
Comment (bogdankolendovskyy) [reply common_examples/simple-ethernet/main/idf_component.yml]: You are right, no need to have it. Removed.

File: ethernet_init/idf_component.yml (hunk 2 above)
Comment (bogdankolendovskyy) [reply ethernet_init/idf_component.yml:1]: Done.

**Comment by kostaond** [comment]
Thanks for the update @bogdankolendovskyy.

**Pull request merged**